### PR TITLE
make monitor start() and stop() async

### PR DESCRIPTION
**Pull request by spali**
⚠️⚠️⚠️ Since we do not accept all types of pull requests and do not want to waste your time. Please be sure that you have read pull request rules:
https://github.com/louislam/uptime-kuma/blob/master/CONTRIBUTING.md#can-i-create-a-pull-request-for-uptime-kuma

Tick the checkbox if you understand [x]: 
- [x] I have read and understand the pull request rules.

# Description

- follow up to #2664
- Preperation for #898 (#680).

- make monitor `start()` and `stop()` async to allow for i.e. db calls.

## Type of change

Please delete any options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- Other

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and tested it
- [x] I have commented my code, particularly in hard-to-understand areas
  (including JSDoc for methods)
- [x] My changes generate no new warnings
- [ ] My code needed automated testing. I have added them (this is optional task)

## Screenshots (if any)

Please do not use any external image service. Instead, just paste in or drag and drop the image here, and it will be uploaded automatically.

### CODE REVIEW

**Comment by chakflying** [comment]
Tested rebased on master seems to work fine. Should be no reason blocking this to merge? @louislam

File: server/server.js [hunk 1]
```diff
@@ -1741,7 +1741,7 @@ async function startMonitors() {
     }
 
     for (let monitor of list) {
-        monitor.start(io);
+        await monitor.start(io);
```
Comment (louislam) [inline server/server.js:1744]: @chakflying Adding `async` to the function is ok, but I am not sure if adding `await` here or other places are ok or not.

For example, one of monitors gets sucked here and the whole application gets sucked by this monitor. Not quite sure.

Comment (spali) [reply server/server.js:1744]: I see your point.
It's the same behavior as before, as it was with synchronous execution which also was blocked by the monitor start. So I see currently no point in changing this now.

In the future, the startup process could be changed to start all monitors async in the background without waiting. But I think this is out of scope for now in this PR.

File: server/model/monitor.js [hunk 2]
```diff
@@ -326,7 +326,7 @@ class Monitor extends BeanModel {
      * @param {Server} io Socket server instance
      * @returns {void}
```
Comment (CommanderStorm) [inline server/model/monitor.js:327]:
```suggestion
     * @returns {Promise<void>}
```

**Comment by CommanderStorm** [comment]
@louislam FYI: I put this in v2.1 as I don't see any blocking change and think this can be merged as is from my side

**Comment by CommanderStorm** [comment]
Thanks for the change.
All changes in this PR are small and uncontroversial (I agree with @spali on this one that refactoring the logic to be less blocking should be done in another PR)
⇒ merging with junior maintainer approval.

**Pull request merged**